include/mesh/triangulator_interface.h

```diff
@@ -73,25 +73,36 @@ class AutoAreaFunction : public FunctionBase<Real>
     std::vector<std::string> field_vars{"f"};
     _auto_area_mfi->set_field_variables(field_vars);
     _auto_area_mfi->get_source_points() = input_pts;
+#ifdef LIBMESH_USE_COMPLEX_NUMBERS
+    std::vector<Number> input_complex_vals;
+    for (const auto & input_val : input_vals)
+      input_complex_vals.push_back(Complex (input_val, 0.0));
+    _auto_area_mfi->get_source_vals() = input_complex_vals;
+#else
     _auto_area_mfi->get_source_vals() = input_vals;
+#endif
     _auto_area_mfi->prepare_for_use();
     this->_initialized = true;
   }
 
   virtual Real operator() (const Point & p,
-                             const Real /*time*/) override
+                           const Real /*time*/) override
   {
     libmesh_assert(this->_initialized);
 
     std::vector<Point> target_pts;
-    std::vector<Real> target_vals;
+    std::vector<Number> target_vals;
 
     target_pts.push_back(p);
     target_vals.resize(1);
 
     _auto_area_mfi->interpolate_field_data(_auto_area_mfi->field_variables(), target_pts, target_vals);
 
+#ifdef LIBMESH_USE_COMPLEX_NUMBERS
+    return target_vals.front().real();
+#else
     return target_vals.front();
+#endif
```

Comment:
We have a `libmesh_real(T)` function that ought to let us turn these 5 lines into 1.

Reply:
I will definitely fix the `libmesh_real(T)` use today.

Let me try to fix header inclusion (I thought it was no longer needed after using the FunctionBase approach). If I cannot fix it by Monday, may be we can merge this first so that the MOOSE part of this feature can be reviewed and merged soon. Thanks.

Reply:
Agreed.  I'm going to be on leave for much of Monday but I'll try to keep an eye out for updates here; I should have time to check around noon.

The header inclusion is only a mild annoyance, but it should be easy to fix - just move the destructor definition (even "= default" counts as a definition) into the .C file, leaving only a declaration in the header, and then we can remove that include and replace it with a forward declaration.

Reply:
Oh!  I should have realized you had other functions that would require more than a forward declaration.  The constructor had to be moved because `std::make_unique` needs to see the constructor of the class inside the `unique_ptr`.

IMHO those functions are complicated enough to belong in the .C file anyway.  Thanks!